src/Console/Util/PhpProcess.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Console\Util;
+
+use Composer\XdebugHandler\XdebugHandler;
+use Symfony\Component\Process\Process;
+
+/**
+ * Vanilla PHP process and utility functions
+ *
+ * @internal
+ */
+final class PhpProcess extends Process
+{
+    /**
+     * @var string|bool
+     */
+    private static $phprc;
+
+    /**
+     * Setups a default Xdebug-free environment for all subprocesses:
+     *
+     * - PHPRC should point to our temporary php.ini (we need to save a previous value)
+     *
+     * - PHP_INI_SCAN_DIR should be made blank because our php.ini has all we need
+     *   (a previous value can be found in XdebugHandler::getRestartSettings()["scanDir"])
+     */
+    public static function setupEnvironment()
+    {
+        if (!isset(self::$phprc)) {
+            self::$phprc = getenv('PHPRC');
+        }
+
+        self::putenv('PHPRC', XdebugHandler::getRestartSettings()['tmpIni']);
+        self::putenv('PHP_INI_SCAN_DIR', '');
+    }
+
+    public function start(callable $callback = null, array $env = null)
+    {
+        // Xdebug wasn't skipped, running as is
+        if ('' == XdebugHandler::getSkippedVersion()) {
+            parent::start($callback, $env);
+
+            return;
+        }
+
+        self::restoreVanillaEnvironment();
```

Comment:
1. Am I right that we need to restore original env here because it was set up by calling `->setupEnvironment()` on `Application.php`?

2. what do you think about `restoreOriginalEnvironment()`? feel free to ignore

3. would you mind adding some comments here about why we restore and then setup xdebug-free env again? I would highly simplify understanding of the whole feature

Reply:
1. That's correct, only we have no choice of *not* calling `setupEnvironment()` in `Application.php`.

2. "Vanilla" is very concrete, "original" is not as much. Where vanilla means the very-very original environment, just an original environment could be some other environment.

3. I'll see to it.